test/controllers/NewUkimsNumberControllerSpec.scala

```diff
@@ -0,0 +1,240 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.SpecBase
+import base.TestConstants.{testEori, userAnswersId}
+import connectors.TraderProfileConnector
+import forms.UkimsNumberFormProvider
+import models.{NormalMode, TraderProfile, UserAnswers}
+import navigation.{FakeNavigator, Navigator}
+import org.mockito.ArgumentMatchers.{any, eq => eqTo}
+import org.mockito.Mockito.when
+import org.scalatestplus.mockito.MockitoSugar
+import pages.UkimsNumberPage
+import play.api.data.FormError
+import play.api.inject.bind
+import play.api.mvc.Call
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import repositories.SessionRepository
+import views.html.NewUkimsNumberView
+
+import scala.concurrent.Future
+
+class NewUkimsNumberControllerSpec extends SpecBase with MockitoSugar {
+
+  private def onwardRoute = Call("GET", "/foo")
+
+  val formProvider = new UkimsNumberFormProvider()
+  private val form = formProvider()
+
+  val mockTraderProfileConnector: TraderProfileConnector = mock[TraderProfileConnector]
+
+  val mockSessionRepository: SessionRepository = mock[SessionRepository]
+
+  when(mockTraderProfileConnector.checkTraderProfile(any())(any())) thenReturn Future.successful(false)
+
+  "NewUkimsNumberController" - {
+
+    ".create" - {
+
+      val newUkimsNumberRoute = routes.NewUkimsNumberController.onPageLoad(NormalMode).url
+
+      "must return OK and the correct view for a GET" in {
+
+        val application = applicationBuilder(userAnswers = Some(emptyUserAnswers))
+          .overrides(
+            bind[Navigator].toInstance(new FakeNavigator(onwardRoute)),
+            bind[TraderProfileConnector].toInstance(mockTraderProfileConnector)
+          )
+          .build()
+
+        running(application) {
+          val request = FakeRequest(GET, newUkimsNumberRoute)
+
+          val result = route(application, request).value
+
+          val view = application.injector.instanceOf[NewUkimsNumberView]
+
+          status(result) mustEqual OK
+          contentAsString(result) mustEqual view(
+            form,
+            routes.NewUkimsNumberController.onSubmit(NormalMode)
+          )(
+            request,
+            messages(application)
+          ).toString
+        }
+      }
+
+      "must populate the view correctly on a GET when the question has previously been answered" in {
+
+        val userAnswers = UserAnswers(userAnswersId).set(UkimsNumberPage, "answer").success.value
```

Comment:
does this need to be New page?

Reply:
This is supposed to pre-populate the form with the exisitng ukims number